### ISSUE TITLE
Sent photos are compressed

### ISSUE THREAD

**Issue by drws** [signalapp/Signal-Android#11477]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

### Bug description

When sending a photo via Signal, it is downsized, recompressed and its EXIF data is removed. The same happens if a photo is sent as a normal file. Signal detects it is a photo and tampers with it nonetheless.

The current implementation effectively removes a great portion of information from the file. Especially rescaling the photo greatly reduces its filesize, but information is lost. All this might be undesired in many cases and I believe it falls under _existing functionality that does not work as intended_.

In the case of EXIF data there is an argument that it should be removed due to privacy concerns, especially the location tag. I think it is a great feature that Signal is able to filter EXIF, but if it is a desirable feature and not a bug, it should be properly implemented and user-configurable (probably with a separate option for location tag filtering).

In the case of photo downsizing, it is especially dire with photos taken with Signal. Even the saved local copies of the photos cannot be saved in higher quality and do not contain any EXIF data. As was previously mentioned, it is also hard to send previously taken photo in full quality. I think photos shouldn't be tampered with by default at least when sending files via generic file transfer function.

**To recap**, this bug report contains three **related** issues:
- Photos, taken via Signal, cannot be saved in full quality and do not contain EXIF data.
- Photos, sent via Signal, are downscaled and EXIF-filtered and even in the case of general file transfer, hacks need to be employed to circumvent this feature, effectively making it a bug.
- EXIF filtering with its current silent implementation leaves a question whether it is a feature or a bug.

**Comment by infinitewaveparticle** [comment]
The quality part I agree with, but by default exif data needs to be stripped. I support a setting to disable exif stripping tho.

**Comment by greyson-signal** [comment]
In general, we do not believe the solution to complex questions like "should I attach EXIF data to this image" is to add more options and put the burden on the user. Signal will always strip out EXIF data by default, and we will always compress photos by default. We added a quality selector not too long ago that lets you choose the photo quality when sending an image, so hopefully that helps.

I think it's worth considering a possible change in behavior if the user selects the image as a file attachment. However, right now things are working as intended, and we'll think more about the file attachment behavior as a possible change. Thanks!

**Comment by drws** [comment]
File-transferring functionality should not tamper with **any files** in **any way,** not just photos. It should provide file integrity during transfer and it is also the most troubled issue discussed. While photo filtering is arguably a feature when using special photo-sending option, it is rather unexpected when using general file transfer.

File attachment behavior change would be a great solution to everything. Then all of the functionality is available while not much new settings are introduced.

There is also a question about being able to keep full-quality (and EXIF-tagged) local copies of photos taken in-app. I hope this issue is considered as well.

And then there's the problem of both compression and EXIF filtering being silently implemented and not even mentioned in the app.

I don't think all of these issues were properly discussed, let alone resolved and I hope the closing of this report will not make it a stale case.

**Comment by infinitewaveparticle** [comment]
I agree about the file attachment idea. My shortcut for sending uncompressed photos is simply changing the file extension to something random and having the recipient change it back to jpg.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by infinitewaveparticle** [comment]
@drws While I understand your opinion there are two issues why I think Signal won't be changing this:

1. Photos and videos make up the bulk of Signal's server bandwidth. While this isn't an issue for smaller, less popular projects, extremely well-funded projects (Telegram), or apps run by huge companies that make loads of money (WhatsApp/Facebook), it IS an issue for Signal - especially as they add more and more users. As of now, while they do bring in some significant donations, the bulk of their operating budget is in the form of a loan that's quickly being depleted. There's no better way to cut down on costs than to make compression of sent images and videos the default. Do I like it? No.

2. Signal is, whether you like it or not, an app focused on privacy and security by default. Users who know what exif data is likely (hopefully) understand the risks of distributing photos with full exif data intact. If you're sending photo albums of your dog to your grandma you'll likely have nothing to worry about (unless you have enemies who are tracking your relatives, your grandma posts the photos online, and your enemies then collect the GPS coordinates of the locations of the photos). However, many users don't understand what this data is (and some - like me - do and appreciate the data being stripped by default), and Signal aims to protect them. Journalists, activists in dictatorships, people with controversial roles, famous people, everyday ordinary people sharing intimate photos, etc, etc, all would appreciate this feature.

I've stated many times that I think customization should exist for every feature - or most features - in every app, and I agree that Signal certainly lacks in customization (they are trying to keep the size of the app as small as possible as well). However, instead of offering this to the bulk of users and certainly instead of offering to keep exif data intact by default, perhaps Signal could offer a cheap paid version (under $10) as a second app with a different name entirely, so as not to confuse new users searching for the original Signal app. That way people who really need such options can choose them without affecting less technical users - and Signal can get a small fee for offering the uptick in bandwidth. Ideal? No. Would I buy it? Probably not.

That's my take, anyway.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by drws** [comment]
@infinitewaveparticle

I understand the bandwidth/storage concern and the effectiveness of the solution. Regarding your second point, Signal's privacy-oriented nature is one of the main reasons I use it. As you can see from the thread, I've also described the issue and possible solutions while emphasizing privacy issues regarding it. And what we've primarily discussed would not really _change_ much, not even the defaults. Unfiltered image files can already be sent by using small hacks. Also raw image transfer could only be accessible as an additional option while using file transfer, not as a default. But again, I think a file transfer tool should not filter files in any way or at least provide a choice to do so, especially not a secure and democratic one.

**Comment by infinitewaveparticle** [comment]
@drws

I agree. Currently I don't mind simply changing the extension of an image or adding several to a zip file if I want to preserve its full quality. Perhaps Signal could add an archive utility so users could send multiple images/files as a compressed zip/far file and the recipient(s) could view them without leaving the app or choose to save them to their devices.

Sent from ProtonMail mobile

-------- Original Message --------
